Updating the K-NN Filters documentation due to recent enhancements in… (#4987)
opensearch-project/documentation-website: _search-plugins/knn/settings.md

## Cluster settings

Setting | Default | Description

> :--- | :-- | :---

Reviewer: ```suggestion
:--- | :--- | :---
```

`knn.memory.circuit_breaker.limit` | 50% | The native memory limit for native library indexes. At the default value, if a machine has 100 GB of memory and the JVM uses 32 GB, the k-NN plugin uses 50% of the remaining 68 GB (34 GB). If memory usage exceeds this value, k-NN removes the least recently used native library indexes.
`knn.memory.circuit_breaker.enabled` | true | Whether to enable the k-NN memory circuit breaker.
`knn.plugin.enabled`| true | Enables or disables the k-NN plugin.

> `knn.model.index.number_of_shards`| 1 | The number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate Nearest Neighbor(ANN) Search.

Reviewer: ```suggestion
`knn.model.index.number_of_shards`| 1 | The number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate Nearest Neighbor (ANN) search.
```

`knn.plugin.enabled`| true | Enables or disables the k-NN plugin.

> `knn.model.index.number_of_shards`| 1 | Number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate k-NN Search.

Reviewer 2: ```suggestion
`knn.model.index.number_of_shards`| 1 | The number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate k-NN Search.
```

> `knn.model.index.number_of_replicas`| 1 | Number of replica shards to use for the model system index. Generally, in a multi-node cluster, this should be at least 1 to increase stability.

Reviewer 2: ```suggestion
`knn.model.index.number_of_replicas`| 1 | The number of replica shards to use for the model system index. Generally, in a multi-node cluster, this should be at least 1 to increase stability.
```

> `knn.advanced.filtered_exact_search_threshold`| null | Threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If number of filtered IDs in a segment is less than this setting's value, exact search will be done on the filtered IDs.  

Reviewer 2: ```suggestion
`knn.advanced.filtered_exact_search_threshold`| null | The threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If the number of filtered IDs in a segment is less than this setting's value, exact search will be performed on the filtered IDs.  
```

> `knn.advanced.filtered_exact_search_threshold`| null | Threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If number of filtered IDs in a segment is less than this setting's value, exact search will be done on the filtered IDs.  

Reviewer 2: Here we use "ANN", above we use "Approximate k-NN Search", and earlier we spelled out "approximate nearest neighbor". Please ensure that we define ANN on first appearance and use only the abbreviation thereafter.

> `knn.advanced.filtered_exact_search_threshold`| null | Threshold value for the filtered ids that will be used to switch to exact search during Filtered ANN Search. If number of filteredIds on a segment is less than the settings value, exact search will be done on the filtered Ids. 

Reviewer: ```suggestion
`knn.advanced.filtered_exact_search_threshold`| null | Threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If number of filtered IDs in a segment is less than this setting's value, exact search will be done on the filtered IDs. 
```
